libraries/AP_HAL_Linux/UARTDevice.cpp

```diff
@@ -0,0 +1,111 @@
+#include <AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+#include <termios.h>
+#include <stdio.h>
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <poll.h>
+
+#include "UARTDevice.h"
+
+#include <termios.h>
+#include <stdio.h>
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <poll.h>
+
+UARTDevice::UARTDevice(char *device_path): 
+    _device_path(device_path)
+{
+}
+
+UARTDevice::~UARTDevice()
+{
+
+}
+
+bool UARTDevice::close()
+{
+    if (::close(_fd) < 0) {
+        return false;
+    }
+
```

Comment:
it would be good to set _fd to -1 after a close(), then check for -1. Otherwise an API error could cause a close of a different fd